com.unity.visualeffectgraph/Editor/ShaderGraph/VFXShaderGraphParticleOutput.cs

```diff
@@ -235,10 +235,27 @@ public override void SetupMaterial(Material material)
             var shaderGraph = GetOrRefreshShaderGraphObject();
             if (shaderGraph != null && shaderGraph.generatesWithShaderGraph)
             {
+                var needsSync = false;
+                var properties = ShaderUtil.GetMaterialProperties(new[] { material });
+                foreach (var property in shaderGraph.properties)
+                {
+                    var orderedProperty = properties.SingleOrDefault(x => x.name == property.referenceName);
+                    if (orderedProperty != null)
+                    {
+                        var order = Array.IndexOf(properties, orderedProperty);
+                        if (order != property.order)
+                        {
+                            property.order = order;
```

Comment:
If we need an other information in AbstractShaderProperty, we should fill them in [ShaderGraphImporter](https://github.com/Unity-Technologies/Graphics/blob/master/com.unity.shadergraph/Editor/Importers/ShaderGraphImporter.cs#L500), if not possible, it should a transient data in VFXShaderGraphParticleOutput.
We aren't suppose to modify the `ShaderGraphVfxAsset` in VFX, this is the ownership of shaderGraph.
Investigating on a better solution 👀

Reply:
Our investigation led to a completely different approach see next commit